frontend/src/components/forms/CreateResident.tsx

```diff
@@ -0,0 +1,260 @@
+import React, { useEffect, useState } from "react";
+import Select, { SingleValue } from "react-select";
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  Box,
+  Button,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Input,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalOverlay,
+  ModalFooter,
+  ScaleFade,
+  Divider,
+} from "@chakra-ui/react";
+
+import { AddIcon } from "@chakra-ui/icons";
+import { SingleDatepicker } from "chakra-dayzed-datepicker";
+import { Card, Col, Row } from "react-bootstrap";
+import CommonAPIClient from "../../APIClients/CommonAPIClient";
+
+import selectStyle from "../../theme/forms/selectStyles";
+import { singleDatePickerStyle } from "../../theme/forms/datePickerStyles";
+
+// TODO: Connect to Buidings table
+const BUILDINGS = [
+  { label: "144", value: "144" },
+  { label: "362", value: "362" },
+  { label: "402", value: "402" },
+];
+
+const CreateResident = (): React.ReactElement => {
+  const [initials, setInitials] = useState("");
+  const [roomNumber, setRoomNumber] = useState("");
+  const [moveInDate, setMoveInDate] = useState(new Date());
+  const [building, setBuilding] = useState("");
+
+  const [initialsError, setInitialsError] = useState(false);
+  const [roomNumberError, setRoomNumberError] = useState(false);
+  const [moveInDateError, setMoveInDateError] = useState(false);
+  const [buildingError, setBuildingError] = useState(false);
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+
+  const addResident = async () => {
+    await CommonAPIClient.createResident({
+      initial: initials.toUpperCase(),
+      roomNum: parseInt(roomNumber, 10),
+      dateJoined: moveInDate,
+      building,
+    });
+  };
+
+  const handleInitialsChange = (e: { target: { value: unknown } }) => {
+    const inputValue = e.target.value as string;
+    if (/^[a-z]{0,2}$/i.test(inputValue)) {
+      setInitials(inputValue.toUpperCase());
+      setInitialsError(false);
+    } else {
+      setInitialsError(true);
```

Comment:
remove this else statement, causing an issue when typing in initials with more than 2 char